race-buffer/src/lib.rs

```diff
@@ -0,0 +1,421 @@
+#![cfg_attr(not(feature = "default"), no_std)]
```

Comment:
The usual idiom in Rust is to have a feature named "std" that, when present, means that the standard library should be in use.